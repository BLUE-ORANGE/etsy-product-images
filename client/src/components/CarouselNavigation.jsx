import React from 'react';
// import PropTypes from 'prop-types';

const CarouselNavigation = props => (
  <ul className="slick-dots">
    {
      props.images.map((image, i) =>
        (
          image.focused ?
            <li key={`a ${image.toString()}`} onClick={() => props.clickHandler(i)}>
              <img src={image.imageUrl} alt="random cat" style={{ width: '30px', height: '30px', borderRadius: '3px' }} />
            </li> :
            <li key={`b ${image.toString()}`} onClick={() => props.clickHandler(i)}>
              <img src={image.imageUrl} alt="random cat" className="thumb-default" style={{ width: '30px', height: '30px', borderRadius: '3px' }} />
            </li>
        ))
    }
  </ul>
);


export default CarouselNavigation;
