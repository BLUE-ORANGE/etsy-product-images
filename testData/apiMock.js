module.exports.testData = [
  {
    id: 108,
    imageUrl: 'https://i.imgur.com/zmTGRu1.jpg',
    focused: true,
    productId: 108,
    createdAt: '2018-02-26T00:17:52.802Z',
    updatedAt: '2018-02-26T00:17:52.802Z',
  }, {
    id: 247,
    imageUrl: 'https://i.imgur.com/V1XNLjh.jpg',
    focused: false,
    productId: 108,
    createdAt: '2018-02-26T00:17:52.802Z',
    updatedAt: '2018-02-26T00:17:52.802Z',
  }, {
    id: 731,
    imageUrl: 'https://i.imgur.com/jpc5JTA.jpg',
    focused: false,
    productId: 108,
    createdAt: '2018-02-26T00:17:52.802Z',
    updatedAt: '2018-02-26T00:17:52.802Z',
  },
];
