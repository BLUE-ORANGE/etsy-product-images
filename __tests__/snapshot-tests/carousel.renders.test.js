import React from 'react';
import renderer from 'react-test-renderer';
import CarouselNavigation from '../../client/src/components/CarouselNavigation';
import testData from '../../testData/apiMock';


it('carousel navigation renders correctly', () => {
  const tree = renderer
    .create(<CarouselNavigation images={testData.testData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


