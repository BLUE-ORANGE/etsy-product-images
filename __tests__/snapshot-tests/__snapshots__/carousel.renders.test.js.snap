// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`carousel navigation renders correctly 1`] = `
<ul
  className="slick-dots"
>
  <li
    onClick={[Function]}
  >
    <img
      alt="random cat"
      src="https://i.imgur.com/zmTGRu1.jpg"
      style={
        Object {
          "borderRadius": "3px",
          "height": "30px",
          "width": "30px",
        }
      }
    />
  </li>
  <li
    onClick={[Function]}
  >
    <img
      alt="random cat"
      className="thumb-default"
      src="https://i.imgur.com/V1XNLjh.jpg"
      style={
        Object {
          "borderRadius": "3px",
          "height": "30px",
          "width": "30px",
        }
      }
    />
  </li>
  <li
    onClick={[Function]}
  >
    <img
      alt="random cat"
      className="thumb-default"
      src="https://i.imgur.com/jpc5JTA.jpg"
      style={
        Object {
          "borderRadius": "3px",
          "height": "30px",
          "width": "30px",
        }
      }
    />
  </li>
</ul>
`;
